src/contract-ui/tabs/edit-extensions/components/ExtensionForm.tsx

```diff
@@ -0,0 +1,406 @@
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { thirdwebClient } from "../../../../@/constants/client";
+import { compatibleExtensions } from "@thirdweb-dev/sdk";
+import { download } from "thirdweb/storage";
+import { resolveImplementation } from "thirdweb/utils";
+import { TransactionButton } from "components/buttons/TransactionButton";
+
+type FormData = {
+  publisherAddress: string;
+  extensionContract: string;
+  version: string;
+};
+
+type InstallExtensionFormProps = {
+  contract: ContractOptions;
+  refetchExtensions: () => void;
+  account: Account;
+  installedExtensions: {
+    data?: string[];
+    isLoading: boolean;
+  };
+};
+
+export const InstallExtensionForm = (props: InstallExtensionFormProps) => {
+  const { register, watch, formState, resetField, reset } = useForm<FormData>({
+    defaultValues: {
+      version: "latest",
+    },
+  });
+  const toast = useToast();
+  const { contract, account } = props;
+  const { errors } = formState;
+
+  const publishedContractsQuery = usePublishedContractsQuery(
+    watch("publisherAddress"),
+    "ModularExtension",
+  );
+  const allVersions = useAllVersions(
+    watch("publisherAddress"),
+    watch("extensionContract"),
+  );
+
+  const installMutation = useMutation({
+    mutationFn: async () => {
+      const installTransaction = installPublishedExtension({
+        contract,
+        chain: contract.chain,
+        client: contract.client,
+        account,
+        extensionName: watch("extensionContract"),
+        publisherAddress: watch("publisherAddress"),
+        version: watch("version"),
+      });
+
+      const txResult = await sendTransaction({
+        transaction: installTransaction,
+        account,
+      });
+
+      await waitForReceipt(txResult);
+    },
+  });
+
+  const onSubmit = async () => {
+    try {
+      await installMutation.mutateAsync();
+
+      props.refetchExtensions();
+
+      toast({
+        variant: "solid",
+        position: "bottom",
+        title: "Extension installed successfully",
+        status: "success",
+        duration: 5000,
+        isClosable: true,
+      });
+
+      // clear form
+      reset({
+        publisherAddress: "",
+        extensionContract: "",
+        version: "latest",
+      });
+    } catch (error) {
+      toast({
+        variant: "solid",
+        position: "bottom",
+        title: "Failed to install extension",
+        status: "error",
+        duration: 5000,
+        isClosable: true,
+      });
+
+      console.error("Error during installation:", error);
+    }
+  };
+
+  const extensionContractInputProps = register("extensionContract", {
+    required: "Extension name is required",
+  });
+
+  const selectedExtension = publishedContractsQuery.data?.find(
+    (x) => x.id === watch("extensionContract"),
+  );
+
+  // Get core contract bytecode
+  const coreContractByteCodeQuery = useQuery({
+    queryKey: ["getBytecode", contract.address],
```

Comment:
missing contract chain id in the query key, meaning if u have the same address on 2 different chains ur gonna share the query result

Reply:
added